### PR TITLE
Imageviewer

### PR DESCRIPTION
**Pull request by alexandercausey**
Added modal for expanded image w/ buttons.
Added click to zoom on expanded image.
Added relative mouse tracking on zoom.

Began implementation of cart api, but currently it's incomplete.

### CODE REVIEW

File: Client/src/Components/product-overview/ImageExpander.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React, { useContext } from "react";
+import PropTypes from "prop-types";
+import DataContext from "../context";
+import css from "./styles.css";
+import { left, right } from "./helper-functions";
+
+function ImageExpander(props) {
```
Comment (ZoyaStudio) [inline Client/src/Components/product-overview/ImageExpander.jsx:7]: you can destructure props within the parenthesis.

File: Client/src/Components/product-overview/ImageExpander.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const { thumb, setThumb, toggleExpanded } = props;
+  const data = useContext(DataContext);
```
Comment (ZoyaStudio) [inline Client/src/Components/product-overview/ImageExpander.jsx:9]: You can desture useContext(DataContext) on line 9.

File: Client/src/Components/product-overview/StyleSelector.jsx [hunk 3]
```diff
@@ -24,6 +26,7 @@ function StyleSelector({ setStyleIndex }) {
     // purpose: provide sizes for selector
     Object.keys(styles[styleIndex].skus).forEach((key) => {
```
Comment (ZoyaStudio) [inline Client/src/Components/product-overview/StyleSelector.jsx:27]: Why not use map here?
const sizes = Object.keys(...).map(..)

File: Client/src/Components/product-overview/StyleSelector.jsx [hunk 4]
```diff
@@ -37,6 +40,13 @@ function StyleSelector({ setStyleIndex }) {
     e.preventDefault();
     setCurrentSelect(e.target.value);
   };
+  const updateQuantity = (e) => {
+    // input: new quantity selected
+    // output: new state
```
Comment (ZoyaStudio) [inline Client/src/Components/product-overview/StyleSelector.jsx:45]: I think output is only for explicit returns

File: Client/src/Components/product-overview/StyleSelector.jsx (hunk 3 above)
Comment (alexandercausey) [reply Client/src/Components/product-overview/StyleSelector.jsx:27]: When I tried to refactor to use map I'm getting errors related to an unfulfilled promise. I am taking a note to revisit to refactor at a later time.

**Comment by alexandercausey** [comment]
Changes made, I'm going to close and resubmit the PR.

**Pull request closed without merging**